queue_and_stack/dll_queue.py

```diff
@@ -7,13 +7,18 @@ class Queue:
     def __init__(self):
         self.size = 0
         # Why is our DLL a good choice to store our elements?
-        # self.storage = ?
+        self.storage = DoublyLinkedList()
 
     def enqueue(self, value):
-        pass
+        self.size += 1
+        self.storage.add_to_tail(value)
 
     def dequeue(self):
-        pass
+        if self.size > 0:
+            self.size -= 1
+            return self.storage.remove_from_head()
+        else:
+            return None
 
     def len(self):
```

Comment:
It's nice that you are keeping track of the length so as to have better control over the data structure well done.
This can be better improved like so
```
def len(self):
        # pass
        # get number of items in stack or queue
        if self.size > 0:
            return self.size
        else:
            return 0
```